platforms/Linux/RPM/Amazonlinux/2023/install_ld.gold.sh

```diff
@@ -0,0 +1,16 @@
+dnf install gmp-devel mpfr-devel texinfo bison git gcc-c++ -y
```

Comment:
Add shebang (#!/bin/bash) for script execution.
Provide descriptive comments for each step, enhancing readability and understanding.